subsys/bluetooth/host/gatt.c

```diff
@@ -1473,6 +1473,11 @@ static int gatt_unregister(struct bt_gatt_service *svc)
 	return 0;
 }
 
+void bt_gatt_unregister_all(void)
+{
+	while (sys_slist_get(&db)) {}
+}
```

Comment:
I guess it makes sense, but is that necessary? 

If we deregister all the dynamically registered GATT services, then we need to implement a way for the services that are registered to know that they have become unregistered. Many of them have persisting data that needs to be cleared, e.g. `bt_vcs_register` has a `static bool registered` that would prevent the service from being registered again.

Reply:
I was under the idea that we should basically clean everything up from bt_disable, but would be open to the idea of leaving these services registered if that makes more sense

Reply:
This is similar to the callbacks:
It does not really serve a purpose to unregister all dynamically registered GATT service. First of, the statically registered are still there, so this only partially works. Secondly, the services, like the callbacks, can be registered regardless of the BT device state. Lastly, if we do want to unregister them (without any reason that I can think of), then we need to somehow notify whomever registered them in the first place, that they are now unregistered. 

I'm not 100% against this, as semantically it makes sense, but it serves no functional purpose besides just adding additional work to register and unregister, as well as additional work for the developer defining and registering the service, as they also would need to deal with the unregister.